### PR TITLE
Adding patch add operation for Related Person Identifier

### PR DESCRIPTION
**Pull request by ahamdany**
Description
----
Adding add operation for Related Person Identifier.

Screen Shot 2021-02-16 at 2 50 15 PM

Screen Shot 2021-02-16 at 2 50 27 PM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/example_json/r4_examples_relatedperson.rb [hunk 1]
```diff
@@ -761,6 +761,26 @@ module Resources
...
+            ]
+          },
+          'system': '2.16.840.1.113883.6.1000',
+          'value': 'THIS:IS:A:UNIQUE:IDENTIFIER',
+          'period': {
+            'start': '2018-01-02T00:00:00-05:00',
+            'end': '2022-01-02T00:00:00-05:00'
+          }
+        }
+      },
```
Comment (exopsRamba) [inline lib/resources/example_json/r4_examples_relatedperson.rb:783]: Should we keep the example here and in the field specific examples the same?

File: lib/resources/r4/related_person_patch.yaml [hunk 2]
```diff
@@ -65,6 +65,40 @@ operations:
...
+            "start": "2016-01-02T00:00:00-05:00",
+            "end": "2020-01-02T00:00:00-05:00"
+          }
+        }
+      }
+    note: >
+      <ul>
+        <li>Must not contain the `use` fields.</li>
+        <li>Must contain the `type`, `system` and `value` fields.</li>
+        <li>If a  time component is included specify timezone when specifying a `period`.</li>
```
Comment (Coffeevee) [inline lib/resources/r4/related_person_patch.yaml:99]: Nit: remove double space between `a  time...`

File: lib/resources/r4/related_person_patch.yaml [hunk 3]
```diff
@@ -65,6 +65,40 @@ operations:
         <li>If `line` is provided, only the first four values are used. Additional values will be ignored.</li>
       </ul>
 
+  - name: add-identifier
+    path: /identifier/-
+    operation: add
+    type: Identifier
+    url: https://hl7.org/fhir/R4/patient-definitions.html#RelatedPerson.identifier
```
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:72]: This should be related-person-definitions.

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (Coffeevee) [inline lib/resources/r4/related_person_patch.yaml:99]: Maybe reword to: "If a time component is included, specify the timezone when specifying a `period`."

File: lib/resources/r4/related_person_patch.yaml [hunk 4]
```diff
@@ -65,6 +65,40 @@ operations:
...
+          "period": {
+            "start": "2016-01-02T00:00:00-05:00",
+            "end": "2020-01-02T00:00:00-05:00"
+          }
+        }
+      }
+    note: >
+      <ul>
+        <li>Must not contain the `use` fields.</li>
+        <li>Must contain the `type`, `system` and `value` fields.</li>
```
Comment (Coffeevee) [inline lib/resources/r4/related_person_patch.yaml:98]: Should we add a comma such that it reads "...the `type`, `system`, and `value`..."? Just a grammatical nit. I prefer the oxford comma, but lmk if we should leave as-is.

File: lib/resources/r4/related_person_patch.yaml [hunk 5]
```diff
@@ -65,6 +65,40 @@ operations:
...
+          "value": "&lt;unique identifier&gt;",
+          "period": {
+            "start": "2016-01-02T00:00:00-05:00",
+            "end": "2020-01-02T00:00:00-05:00"
+          }
+        }
+      }
+    note: >
+      <ul>
+        <li>Must not contain the `use` fields.</li>
```
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:97]: nit: fields -> field.

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (exopsRamba) [inline lib/resources/r4/related_person_patch.yaml:99]: Let's make this list item: `When specifying a period, the fields must include a time component with a timezone.`

Comment (Coffeevee) [reply lib/resources/r4/related_person_patch.yaml:99]: Ignore this comment and focus on Keifer's here: https://github.com/cerner/fhir.cerner.com/pull/555#discussion_r577215581

File: lib/resources/r4/related_person_patch.yaml (hunk 3 above)
Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: [fixed](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR72)

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: [updated](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR99)

File: lib/resources/r4/related_person_patch.yaml (hunk 5 above)
Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: [changed](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL97)

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: updated to [this](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL99)

Comment (Coffeevee) [reply lib/resources/r4/related_person_patch.yaml]: Comment became moot when Ali updated the wording here: https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dR99

Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: Changed it to [this](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL99)

File: lib/resources/example_json/r4_examples_relatedperson.rb (hunk 1 above)
Comment (ahamdany) [reply lib/resources/example_json/r4_examples_relatedperson.rb:783]: yea good idea changed [here](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-133ac83b73a8a88b430a777cf1700b9010258bd96bc4ff69a51350091f73643eL779) and [here](https://github.com/cerner/fhir.cerner.com/pull/555/commits/1530f2ab8911c92413e05c79ec4bf854aa3d1512#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL87)

File: lib/resources/r4/related_person_patch.yaml (hunk 4 above)
Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: hmmm I think I prefer it the way it is but since there are two votes for this I changed it [here](https://github.com/cerner/fhir.cerner.com/pull/555/commits/ada4136825e1bad402aebcac6562da87c1340e33#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL98)

File: lib/resources/r4/related_person_patch.yaml [hunk 6]
```diff
@@ -65,6 +65,40 @@ operations:
...
+            "start": "2016-01-02T00:00:00-05:00",
+            "end": "2020-01-02T00:00:00-05:00"
+          }
+        }
+      }
+    note: >
+      <ul>
+        <li>Must not contain the `use` field.</li>
+        <li>Must contain the `type`, `system`, and `value` fields.</li>
+        <li>When specifying a period, the fields must include a time component with a timezone.</li>
```
Comment (Coffeevee) [inline lib/resources/r4/related_person_patch.yaml:99]: nit: you can add back in the `period` formatting you had before like you are doing for `use`, `type`, etc. above.

Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: good catch, updated [here](https://github.com/cerner/fhir.cerner.com/pull/555/commits/60fbd8b256ad4603508b7cc18c9da8700d58d365#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL99)

File: lib/resources/r4/related_person_patch.yaml [hunk 7]
```diff
@@ -65,6 +65,40 @@ operations:
...
+            "start": "2016-01-02T00:00:00-05:00",
+            "end": "2020-01-02T00:00:00-05:00"
+          }
+        }
+      }
+    note: >
+      <ul>
+        <li>Must not contain the `use` field.</li>
+        <li>Must contain the `type`, `system`, and `value` fields.</li>
+        <li>When specifying a `period`, the fields must include a time component with a timezone.</li>
```
Comment (Coffeevee) [inline lib/resources/r4/related_person_patch.yaml:99]: After some discussion with Ali, we resolved that it might be better to say '... the `start` and `end` fields must include a time component..."

Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: updated [here](https://github.com/cerner/fhir.cerner.com/pull/555/commits/7a379c8b2e991ec3e5b97b19a0035caf7a4f33c8)

**Review by Coffeevee — APPROVED** [review]

File: lib/resources/r4/related_person_patch.yaml (hunk 2 above)
Comment (exopsRamba) [reply lib/resources/r4/related_person_patch.yaml]: Period should still have ` around it.

Comment (ahamdany) [reply lib/resources/r4/related_person_patch.yaml]: fixed as part of this [comment](https://github.com/cerner/fhir.cerner.com/pull/555#discussion_r577225614) here is the [commit](https://github.com/cerner/fhir.cerner.com/pull/555/commits/60fbd8b256ad4603508b7cc18c9da8700d58d365#diff-a0c01bda612e4656e850590d65d36528f1caec8a70317c2ae4cb71d8095c9e7dL99)

**Review by exopsRamba — APPROVED** [review]

**Review by alexbeckner1 — APPROVED** [review]

**Review by MadhurRaj — APPROVED** [review]

**Comment by sydneanschutz** [comment]
Validated on live:
Screen Shot 2021-02-18 at 1 47 44 PM
